content/blog/2020-04-dplyr-1-0-0-rowwise/index.Rmarkdown

```diff
@@ -0,0 +1,205 @@
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(collapse = TRUE, comment = "#>")
+```
+
+This post is the latest in a series of post leading up the the dplyr 1.0.0 release. So far, the series has covered:
+
+* [Major lifecycle changes](https://www.tidyverse.org/blog/2020/03/dplyr-1-0-0-is-coming-soon/).
+* [New `summarise()` features](https://www.tidyverse.org/blog/2020/03/dplyr-1-0-0-summarise/).
+* [`select()`, `rename()`, `relocate()`](https://www.tidyverse.org/blog/2020/03/dplyr-1-0-0-select-rename-relocate/).
+* [Working `across()` columns](https://www.tidyverse.org/blog/2020/04/dplyr-1-0-0-colwise/).
+
+Today, I wanted to talk a little bit about the renewed `rowwise()` function that makes it easy to perform operations "row-by-row". I'll show how you can use `rowwise()` to compute summaries "by row", talk about how `rowwise()` is a natural pairing with list-columns, and show a couple of use cases that I think are particularly elegant. You can learn more about all of these topics in  [`vignette("rowwise")`](https://dplyr.tidyverse.org/dev/articles/rowwise.html).
+
+### Getting the dev version
+
+If you're interested in living life on the edge (or trying out anything you see in this blog post), you can install the development version of dplyr with:
+
+```{r, eval = FALSE}
+devtools::install_github("tidyverse/dplyr")
+```
+
+Note that the development version won't become 1.0.0 until it's released, but it has all the same features.
+
+```{r setup}
+library(dplyr, warn.conflicts = FALSE)
+```
+
+## Basic operation
+
+`rowwise()` works like `group_by()` in the sense that it doesn't change what the data looks like; it changes how dplyr verbs operate on it. Let's see how this works with a simple example. Here I have some imaginary test results for students in a class:
+
+```{r}
+df <- tibble(
+  student_id = 1:4, 
+  test1 = 10:13, 
+  test2 = 20:23, 
+  test3 = 30:33, 
+  test4 = 40:43
+)
+df
+```
+
+It'd like to be able to compute the mean of the test scores for each student, but `mutate()` and `mean()` do what I want:
+
+```{r}
+df %>% mutate(avg = mean(c(test1, test2, test3, test4)))
+```
+
+The problem is that I'm getting a mean over the whole data frame, not for each student. I can resolve this problem a get a mean for each student by creating create a "row-wise" data frame with `rowwise()`:
+
+```{r}
+rf <- rowwise(df, student_id)
+```
+
+`rowwise()` doesn't need any additional arguments unless you have variables that identify the rows, like `student_id` here. Much like grouping variables, they will be automatically preserved when you `summarise()` the data.
+
+```{r}
+rf
+```
+
+`rf` _looks_ very similar to `df`, but _behaves_ very differently:
+
+```{r}
+rf %>% mutate(avg = mean(c(test1, test2, test3, test4)))
+```
+An additional advantage of `rowwise()` is that it's paired with `c_across()`, which works like `c()` but uses the same tidyselect syntax as `across()`. That makes it easy to operate on multiple variables:
```

Comment:
```suggestion
An additional advantage of `rowwise()` is that it's paired with [`c_across()`](https://dplyr.tidyverse.org/dev/reference/across.html), which works like `c()` but uses the same tidyselect syntax as `across()`. That makes it easy to operate on multiple variables:
```

Reply:
I know we don't consistently link to the function references, but I think it's handy here.